### PR TITLE
Discuss Sidebar: implement default avatar for channels

### PR DESCRIPTION
**Pull request by qho-odoo**
- if the avatar is uploaded by a user
	* avatar will be used no matter how other settings are
	* ThreadIcon will display
- if the avatar is not set manually
	* avatar will be computed by `public` and `email_sent`
	* ThreadIcon will hide

### CODE REVIEW

File: addons/mail/models/mail_channel.py [hunk 1]
```diff
@@ -48,7 +44,10 @@ def _get_default_image(self):
     is_chat = fields.Boolean(string='Is a chat', compute='_compute_is_chat')
     description = fields.Text('Description')
     email_send = fields.Boolean('Send messages by email', default=False)
-    image_128 = fields.Image("Image", max_width=128, max_height=128, default=_get_default_image)
+    image_128 = fields.Image("Image", max_width=128, max_height=128, compute='_compute_image_128', inverse='_inverse_image_128')
+    avatar_default = fields.Image("Default avatar", max_width=128, max_height=128, compute='_compute_avatar_default')
+    avatar_upload = fields.Image("User upload avatar")
+    is_default_avatar = fields.Boolean("Is using default avatar?", compute='_compute_is_default_avatar')
```
Comment (seb-odoo) [inline addons/mail/models/mail_channel.py:50]: `image_128` should stay the basic stored field by convention.
And there should only be one extra field, which is a compute `image_or_placeholder_128` (name to be determined later)

Either the `image_128` is set, or it's not. The boolean is not necessary.

And having a field for the default is not necessary either, it should just be used appropriate as part of the placeholder compute method.

File: addons/mail/models/mail_channel.py [hunk 2]
```diff
@@ -147,6 +146,32 @@ def _compute_moderation_count(self):
...
+    @api.depends('public', 'email_send')
+    def _compute_avatar_default(self):
+        for channel in self:
+            if channel.email_send:
+                image_path = modules.get_module_resource('mail', 'static/src/img', 'channel_mailing.png')
+            elif channel.public == 'private':
+                image_path = modules.get_module_resource('mail', 'static/src/img', 'channel_private.png')
+            else:
+                image_path = modules.get_module_resource('mail', 'static/src/img', 'channel.png')
+            channel.avatar_default = base64.b64encode(open(image_path, 'rb').read())
```
Comment (seb-odoo) [inline addons/mail/models/mail_channel.py:163]: Use `file_open` from tools instead, can do all of those step with an easy syntax.

**Review by seb-odoo — COMMENTED** [review]
quick comments about code below, but we discussed the spec with CPE and we will probably change it, because the current result is too inconsistent with the indicator sometimes part of the image and sometimes not

**Comment by qho-odoo** [comment]
PR closed because of changed task requirements.
Check new PR #999

**Pull request closed without merging**